File: tema/src/main/java/ro/mpp/repository/UserDBRepository.java
package ro.mpp.repository;

import ro.mpp.domain.User;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.Properties;

public class UserDBRepository implements UserRepository {

    private JdbcUtils dbUtils;

    private static final Logger logger= LogManager.getLogger();

    public UserDBRepository(Properties props) {
        this.dbUtils = new JdbcUtils(props);
    }

    @Override
    public Optional<User> findOne(Integer integer) {
        return Optional.empty();
    }

    @Override
    public Iterable<User> findAll() {
        logger.traceEntry();
        Connection con=dbUtils.getConnection();
        List<User> users=new ArrayList<>();
        try (PreparedStatement preStmt=con.prepareStatement("select * from users")) {
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    Integer id = result.getInt("id");
                    String username = result.getString("username");
                    String password = result.getString("password");
                    User user = new User(username, password);
                    user.setId(id);
                    users.add(user);
                }
            }
        } catch (SQLException e) {

            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(users);
        return users;
    }

    @Override
    public void save(User entity) {
        logger.traceEntry( "saving task {}",entity);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("insert into users (username, password) values (?,?)")) {
            preStmt.setString(1,entity.getUsername());
            preStmt.setString(2,entity.getPassword());
            int result=preStmt.executeUpdate();
            logger.trace("Saved () instances", result);
        }catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB " + ex);
            logger.traceExit();
        }
    }

    @Override
    public void delete(Integer integer) {
    }

    @Override
    public void update(User entity) {

    }

    @Override
    public User findByUsername(String username) {
        logger.traceEntry();
        User user = null;
        Connection con=dbUtils.getConnection();
        try (PreparedStatement preStmt=con.prepareStatement("select * from users where username=?")) {
            preStmt.setString(1, username);
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    Integer id = result.getInt("id");
                    String password = result.getString("password");
                    user =  new User(username, password);
                    user.setId(id);
                }
            }
        } catch (SQLException e) {

            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(user);
        return user;
    }
}

File: Anul_2/Semestrul_2/Medii de proiectare si programare/mpp-proiect-csharp-andrea-muresan/travelAgency/travelAgency/repository/UserDBRepository.cs
using System;
using System.Collections.Generic;
using System.Data;
using travelAgency.domain;
using travelAgency.utils;

namespace travelAgency.repository
{
    public class UserDBRepository : IUserRepository
    {
        private static readonly log4net.ILog log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
        IDictionary<String, string> props;
        private Encrypt _encrypt = new Encrypt();
        
        public UserDBRepository(IDictionary<String, string> props)
        {
            log.Info("Creating UserDBRepository");
            this.props = props;
        }

        public IEnumerable<User> FindAll()
        {
            throw new NotImplementedException();
        }

        public User? FindOne(int id)
        {
            throw new NotImplementedException();
        }

        public void Save(User entity)
        {
            log.InfoFormat("Saving user values: {0}, {1}", entity.Username, "###");
            IDbConnection con = DBUtils.getConnection(props);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText =
                    "INSERT INTO users(username, password) " +
                    "VALUES (@username, @password)";
                var paramUsername = comm.CreateParameter();
                paramUsername.ParameterName = "@username";
                paramUsername.Value = entity.Username;
                comm.Parameters.Add(paramUsername);
                
                
                var paramPassword = comm.CreateParameter();
                paramPassword.ParameterName = "@password";
                paramPassword.Value = _encrypt.EncodePasswordToBase64(entity.Password);
                comm.Parameters.Add(paramPassword);
                

                int noVal = comm.ExecuteNonQuery();
                log.InfoFormat("{0} users saved", noVal);
            }
        }

        public void Delete(int id)
        {
            throw new NotImplementedException();
        }

        public void Update(User entity)
        {
            throw new NotImplementedException();
        }

        public User FindByUsername(string username)
        {
            log.InfoFormat("Entering FindByUsername with value {0}", username);
            IDbConnection con = DBUtils.getConnection(props);
            
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select id, username, password from users where username=@username";
                IDbDataParameter paramUsername = comm.CreateParameter();
                paramUsername.ParameterName = "@username";
                paramUsername.Value = username;
                comm.Parameters.Add(paramUsername);
            
                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        int idV = dataR.GetInt32(0);
                        String usernameV = dataR.GetString(1);
                        String passwordV = dataR.GetString(2);
                        User user = new User(usernameV, passwordV);
                        user.Id = idV;
                        log.InfoFormat("Exiting findByUsername with value {0}", user);
                        return user;
                    }
                }
            }
            
            log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }
    }
}